core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -259,99 +250,113 @@ oppia.controller('StateResponses', [
+              $scope.unresolvedAnswers.splice(answerIndex, 1);
+              if (answer.classificationResult.classificationCategorization ===
+                  DEFAULT_OUTCOME_CLASSIFICATION) {
+                TrainingDataService.trainDefaultResponse(answer.answer);
+                return;
+              }
+
+              TrainingDataService.trainAnswerGroup(
+                answer.classificationResult.answerGroupIndex, answer.answer);
+            };
+
+            $scope.finishTrainingAnswer = function() {
+              $scope.unresolvedAnswers.splice($scope.selectedAnswerIndex, 1);
+            };
+
+            $scope.openTrainUnresolvedAnswerModal = function(answerIndex) {
+              $scope.selectedAnswerIndex = answerIndex;
+              answer = $scope.unresolvedAnswers[answerIndex].answer;
+              return TrainingModalService.openTrainUnresolvedAnswerModal(
+                answer, true, $scope.finishTrainingAnswer);
+            };
 
             $scope.finishTeaching = function(reopen) {
               $uibModalInstance.close({
                 reopen: reopen
               });
             };
 
-            $scope.submitAnswer = function(answer) {
-              $scope.answerTemplate = (
-                ExplorationHtmlFormatterService.getAnswerHtml(
-                  answer, stateInteractionIdService.savedMemento,
-                  stateCustomizationArgsService.savedMemento));
-
-              var classificationResult = (
-                AnswerClassificationService.getMatchingClassificationResult(
-                  _explorationId, _stateName, _state, answer, rulesService));
-              var feedbackHtml = 'Nothing';
-              var dest = classificationResult.outcome.dest;
-              if (classificationResult.outcome.hasNonemptyFeedback()) {
-                feedbackHtml = classificationResult.outcome.feedback.getHtml();
-              }
-              if (dest === _stateName) {
-                dest = '<em>(try again)</em>';
-              }
-              $scope.trainingDataAnswer = answer;
-              $scope.trainingDataFeedback = feedbackHtml;
-              $scope.trainingDataOutcomeDest = dest;
-
-              var classificationCategorization = (
-                classificationResult.classificationCategorization);
-
-              // If answer is classified using explicit rules then we don't show
-              // yes / no option to creator. Otherwise we ask whether the
-              // response it satisfactory and assign a different response
-              // if response is not satisfactory.
-              if (classificationCategorization === EXPLICIT_CLASSIFICATION) {
-                $scope.classification.answerGroupIndex = -1;
-              } else {
-                $scope.classification.answerGroupIndex = (
-                  classificationResult.answerGroupIndex);
-              }
-            };
+            $scope.showLoadingDots = true;
```

Comment:
$scope.loadingDotsAreShown (to avoid starting with a verb when the variable does not represent a function)